launcher/agent/agent.go

```diff
@@ -165,27 +167,40 @@ func (a *agent) attest(nonce []byte, cel []byte) (*pb.Attestation, error) {
 // It will reset the container image signatures for now.
 func (a *agent) Refresh(ctx context.Context) error {
 	if a.launchSpec.Experiments.EnableSignedContainerCache {
-		signatures := fetchContainerImageSignatures(ctx, a.sigsFetcher, a.launchSpec.SignedImageRepos, a.logger)
+		signatures := fetchContainerImageSignatures(ctx, a.sigsFetcher, a.launchSpec.SignedImageRepos, defaultRetryPolicy(), a.logger)
 		a.sigsCache.set(signatures)
 		a.logger.Printf("Refreshed container image signature cache: %v\n", signatures)
 	}
 	return nil
 }
 
-func fetchContainerImageSignatures(ctx context.Context, fetcher signaturediscovery.Fetcher, targetRepos []string, logger *log.Logger) []oci.Signature {
+func fetchContainerImageSignatures(ctx context.Context, fetcher signaturediscovery.Fetcher, targetRepos []string, retry backoff.BackOff, logger *log.Logger) []oci.Signature {
 	signatures := make([][]oci.Signature, len(targetRepos))
 
 	var wg sync.WaitGroup
 	for i, repo := range targetRepos {
 		wg.Add(1)
 		go func(targetRepo string, index int) {
 			defer wg.Done()
-			sigs, err := fetcher.FetchImageSignatures(ctx, targetRepo)
+
+			// backoff independently per repo
+			var sigs []oci.Signature
+			err := backoff.RetryNotify(
+				func() error {
+					s, err := fetcher.FetchImageSignatures(ctx, targetRepo)
+					sigs = s
```

Comment:
nit: can omit this assign

Reply:
This one is needed as its the only place we actually get the return value from FetchImageSignatures.

Verified by removing it and watching tests fail